### ISSUE TITLE
Analyzer can't analyzer my project - GET_NAVIGATION_INVALID_FILE

### ISSUE THREAD

**Issue by zoechi** [dart-lang/sdk#26190]
See especially the last few lines 

> 2016-04-05 08:43:29,023 [80460169]   INFO - watcher.FwProjectChangeTracker - [tracking vfs changes] stopped 
2016-04-05 08:43:29,023 [80460169]   INFO - watcher.FwProjectChangeTracker - [tracking unsaved documents] stopped 
2016-04-05 08:43:35,169 [80466315]   INFO - ellij.project.impl.ProjectImpl - 104 project components initialized in 174 ms 
2016-04-05 08:43:35,177 [80466323]   INFO - le.impl.ModuleManagerComponent - 1 module(s) loaded in 8 ms 
2016-04-05 08:43:35,228 [80466374]   INFO - roject.impl.ProjectManagerImpl - FW/roots waiting started 
2016-04-05 08:43:35,369 [80466515]   INFO - roject.impl.ProjectManagerImpl - FW/roots waiting finished 
2016-04-05 08:43:35,389 [80466535]   INFO - .diagnostic.PerformanceWatcher - Pushing properties took 2ms; general responsiveness: ok; EDT responsiveness: ok 
2016-04-05 08:43:35,390 [80466536]   INFO - .diagnostic.PerformanceWatcher - Indexable file iteration took 1ms; general responsiveness: ok; EDT responsiveness: ok 
2016-04-05 08:43:35,638 [80466784]   INFO - watcher.FwProjectChangeTracker - [tracking vfs changes] started 
2016-04-05 08:43:35,638 [80466784]   INFO - watcher.FwProjectChangeTracker - [tracking unsaved documents] started 
2016-04-05 08:43:35,675 [80466821]   INFO - tartup.impl.StartupManagerImpl - /home/zoechi/dart/playground/angular2/.idea case-sensitivity: expected=true actual=true 
2016-04-05 08:43:35,690 [80466836]   INFO -       #git4idea.repo.GitConfig - No branch.master.merge/rebase item in the .git/config 
2016-04-05 08:43:35,719 [80466865]   INFO - lij.tasks.impl.TaskManagerImpl - Updating issues cache (every 20 min) 
2016-04-05 08:43:36,087 [80467233]   INFO -       #git4idea.repo.GitConfig - No branch.master.merge/rebase item in the .git/config 
2016-04-05 08:43:36,273 [80467419]   INFO - tor.impl.FileEditorManagerImpl - Project opening took 1280 ms 
2016-04-05 08:43:48,072 [80479218]   INFO -       #git4idea.repo.GitConfig - No branch.master.merge/rebase item in the .git/config 
2016-04-05 08:44:12,507 [80503653]   INFO - yzer.DartAnalysisServerService - Error from analysis_getNavigation(), file = /home/zoechi/dart/playground/angular2/so36397535_interval/lib/app_element.dart, SDK version = 1.16.0-edge.9fc0e5e3910fdf321f30ceff4c954b63faf32a10, server version = 1.15.0, error code = GET_NAVIGATION_INVALID_FILE: Error during `analysis.getNavigation`: invalid file. 
2016-04-05 08:44:12,510 [80503656]   INFO - yzer.DartAnalysisServerService - Error from analysis_getNavigation(), file = /home/zoechi/dart/playground/angular2/so36397535_interval/lib/app_element.dart, SDK version = 1.16.0-edge.9fc0e5e3910fdf321f30ceff4c954b63faf32a10, server version = 1.15.0, error code = GET_NAVIGATION_INVALID_FILE: Error during `analysis.getNavigation`: invalid file. 
2016-04-05 08:44:12,512 [80503658]   INFO - yzer.DartAnalysisServerService - Error from analysis_getNavigation(), file = /home/zoechi/dart/playground/angular2/so36397535_interval/lib/app_element.dart, SDK version = 1.16.0-edge.9fc0e5e3910fdf321f30ceff4c954b63faf32a10, server version = 1.15.0, error code = GET_NAVIGATION_INVALID_FILE: Error during `analysis.getNavigation`: invalid file. 

> Dart VM version: 1.16.0-edge.706ea554049c5809ede4126b9baa654fb4d71bbd (Mon Apr  4 16:52:49 2016) on "linux_x64"

When I open the project in WebStorm, the progress indicator shows DartAnalyzer is working but the Analyzer view never becomes available. In the logs I get above entries.

This is with this folder opened in WebStorm https://github.com/bwu-dart-playground/dart_playground/tree/612cb892838c8b15b24967fe8b48e02c89da8108/angular2

**Comment by zoechi** [comment]
Moved the `angular2` directory to its own GitHub repo https://github.com/bwu-dart-playground/angular2/tree/master (still the same issue)

**Comment by zoechi** [comment]
When I open the directory https://github.com/bwu-dart-playground/angular2/tree/b76998e1d8f461a8e95ded03f0c2c1a88ef35625/so36397535_interval Dart Analysis seems never to complete even though the project consists only of 2 dart and 1 html and the Angular2 dependency.

There are no additional related logs in the `idea.log` file.

**Comment by bwilkerson** [comment]
@jwren

**Comment by jwren** [comment]
I haven't see this before.  Did you see after an SDK or WebStorm upgrade? @alexander-doroshko

**Comment by zoechi** [comment]
I haven's seen this again since I reported the issue.

**Comment by alexander-doroshko** [comment]
`Error during analysis.getNavigation: invalid file` in log may happen when you search for usages of a Dart file. Do you remember what you did when you got a problem? In some cases it is normal.

If Dart Analysis spinner hangs forever - it is never normal.

Dart Analysis tool window makes itself visible by default, but if you minimized it, next time you open project it will be minimized as well. You can access tool windows via main menu or by clicking an icon in IDE bottom-left corner.

If the issue is not reproducible I'd suggest to close it.

**Comment by MikeMitterer** [comment]
I have the same problem.
`2016-07-06 12:21:11,993 [ 547558]   INFO - yzer.DartAnalysisServerService - Error from analysis_getNavigation(), file = /Volumes/Daten/DevLocal/DevDart/BeanValidator.Dart/lib/constraints.dart, SDK version = 1.17.1, server version = 1.15.0, error code = GET_NAVIGATION_INVALID_FILE: Error during `analysis.getNavigation`: invalid file. 
`
dartanalyzer on the commandline works (as always!!!)

**Comment by alexander-doroshko** [comment]
@MikeMitterer Info in log is not always a problem itself. Is there any functionality broken? 
Such issues can be worked on only if we manage to reproduce it. Sample project + steps to reproduce are welcome.

**Comment by sid-kurias** [comment]
I too saw this problem when using emacs. No functionality worked when it happened. I do use a modified version of dart-mode https://github.com/sid-kurias/dart-mode and company-dart https://github.com/sid-kurias/company-dart.

I was using dart sdk 1.15.x, with everything working sometime ago. I hadn't done any dart work recently. Yesterday I cloned the mojito repo https://pub.dartlang.org/packages/mojito and nothing seemed to work, No hierarchy, no completion, no navigation,outline... nothing. The same problem after upgrading the sdk to the latest version 1.17.1.

So I tried to create a new project with stagehand. And everything works with that newly created project. But not with the cloned mojito project. Then I did a git pull on the copy of the shelf project and that seems to work fine. At least in my case it seems to be project specific.
